infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -179,72 +297,103 @@ class FrameConverterServiceV1 @Autowired constructor(
         return translation.t("enum.location-track-type.${locationTrack.type}")
     }
 
-    private fun createCoordinateToTrackMeterResponse(
+    private fun createCoordinateToTrackAddressResponse(
         layoutContext: LayoutContext,
-        request: ValidCoordinateToTrackMeterRequestV1,
+        request: ValidCoordinateToTrackAddressRequestV1,
         nearestMatch: NearestLocationTrackPointMatch,
         trackNumber: TrackNumber,
         geocodedAddress: AddressAndM,
-    ): List<CoordinateToTrackMeterResponseV1> {
+    ): List<CoordinateToTrackAddressResponseV1> {
+        val featureGeometry = createFeatureGeometry(request.responseSettings, nearestMatch.closestPointOnTrack)
 
-        val featureGeometry =
-            if (FrameConverterResponseSettingV1.FeatureGeometry in request.responseSettings) {
-                createFeatureGeometry(nearestMatch)
-            } else {
-                GeoJsonGeometryPoint.empty()
-            }
-
-        val closestLocationTrackMatch =
-            if (FrameConverterResponseSettingV1.BasicFeatureMatchData in request.responseSettings) {
-                createBasicFeatureMatchData(nearestMatch)
-            } else {
-                null
-            }
+        val featureMatchDataSimple = createBasicFeatureMatchDataOrNull(
+            request.responseSettings,
+            nearestMatch.closestPointOnTrack,
+            nearestMatch.distanceToClosestPoint,
+        )
 
-        val conversionDetails =
-            if (FrameConverterResponseSettingV1.FeatureMatchDetails in request.responseSettings) {
-                createDetailedFeatureMatchData(layoutContext, nearestMatch, trackNumber, geocodedAddress)
-            } else {
-                null
-            }
+        val conversionDetails = createDetailedFeatureMatchDataOrNull(
+            request.responseSettings,
+            layoutContext,
+            nearestMatch.locationTrack,
+            trackNumber,
+            geocodedAddress.address,
+        )
 
         return listOf(
-            CoordinateToTrackMeterResponseV1(
+            CoordinateToTrackAddressResponseV1(
                 geometry = featureGeometry,
-                properties = CoordinateToTrackMeterResponsePropertiesV1(
-                    id = request.identifier,
-                    closestLocationTrackMatch = closestLocationTrackMatch,
-                    conversionDetails = conversionDetails,
+                properties = CoordinateToTrackAddressResponsePropertiesV1(
+                    identifier = request.identifier,
+                    featureMatchDataSimple = featureMatchDataSimple,
+                    featureMatchDataDetails = conversionDetails,
                 )
             )
         )
     }
 
-    private fun createDetailedFeatureMatchData(
+    private fun createTrackAddressToCoordinateResponse(
         layoutContext: LayoutContext,
-        nearestMatch: NearestLocationTrackPointMatch,
```

Comment:
Menee diffi aika hassuksi tästä alaspäin, sekoittaa ns. aiemman funkkarin määritelmää uuteen